modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/nhttp/config/ClientConnFactoryBuilder.java

```diff
@@ -341,15 +349,120 @@ private SSLContext createSSLContext(OMElement keyStoreElt, OMElement trustStoreE
                 if (fis != null) {
                     try {
                         fis.close();
-                    } catch (IOException ignore) {}
+                    } catch (IOException ignore) {
+                    }
                 }
             }
         } else if (novalidatecert) {
             if (log.isWarnEnabled()) {
                 log.warn(name + " Server certificate validation (trust) has been disabled. " +
-                    "DO NOT USE IN PRODUCTION!");
-            }            
-            trustManagers = new TrustManager[] { new NoValidateCertTrustManager() };
+                        "DO NOT USE IN PRODUCTION!");
+            }
+            trustManagers = new TrustManager[]{new NoValidateCertTrustManager()};
+        }
+
+        try {
+            final Parameter sslpParameter = transportOut.getParameter("SSLProtocol");
+            final String sslProtocol = sslpParameter != null ? sslpParameter.getValue().toString() : "TLS";
+            SSLContext sslcontext = SSLContext.getInstance(sslProtocol);
+            sslcontext.init(keymanagers, trustManagers, null);
+            return sslcontext;
+
+        } catch (GeneralSecurityException gse) {
+            log.error(name + " Unable to create SSL context with the given configuration", gse);
+            throw new AxisFault("Unable to create SSL context with the given configuration", gse);
+        }
+    }
+
+    private SSLContext createSSLContext(OMElement keyStoreElt, OMElement trustStoreElt,
+                                        boolean novalidatecert, SecretResolver secretResolver) throws AxisFault {
+
+        KeyManager[] keymanagers = null;
+        TrustManager[] trustManagers = null;
+
+
+        if (keyStoreElt != null) {
+            String location = keyStoreElt.getFirstChildWithName(new QName("Location")).getText();
+            String type = keyStoreElt.getFirstChildWithName(new QName("Type")).getText();
+            String storePassword = getSecureVaultValue(secretResolver, keyStoreElt.getFirstChildWithName(new QName
+                    ("Password")));
+            String keyPassword = getSecureVaultValue(secretResolver, keyStoreElt.getFirstChildWithName(new QName
+                    ("KeyPassword")));
+
+            FileInputStream fis = null;
+            try {
+                KeyStore keyStore = KeyStore.getInstance(type);
+                fis = new FileInputStream(location);
```

Comment:
Shall we use [1]
[1] https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
since the input stream is closed correctly, I think my original code would be fine as well